docs/dev/guidelines/server.rst

```diff
@@ -350,3 +350,101 @@ Please read `the AssertJ documentation <https://assertj.github.io/doc/#assertj-c
 
 
 Some parts of these guidelines are adapted from https://medium.com/@madhupathy/ultimate-clean-code-guide-for-java-spring-based-applications-4d4c9095cc2a
+
+22. General Testing Tips
+========================
+
+Write meaningful comments for your tests.
+These comments should contain information about what is tested specifically.
+
+.. code-block:: java
+
+    /**
+     * Tests that borrow() in Book successfully sets the available attribute to false
+     */
+    @Test
+    public void testBorrowInBook() {
+        // Test Code
+    }
+
+Use appropriate and descriptive names for test cases. This makes it easier for other developers to understand what you actually test without looking deeper into it.
+This is the same reason as to why you should not name your variables int a, double b, String c, and so on. For example, if you want to test the method borrow in the class Book, testBorrowInBook() would be an appropriate name for the test case.
+
+.. code-block:: java
+
+    @Test
+    public void testBorrowInBook() {
+        // Test Code
+    }
+
+Try to follow the best practices for java Testing:
+
+* Write small and specific tests by heavily using helper functions, parameterized tests, AssertJ’s powerful assertions, not overusing variables, asserting only what’s relevant and avoiding one test for all corner cases.
+* Write self-contained tests by revealing all relevant parameters, insert data right in the test and prefer composition over inheritance.
+* Write dumb tests by avoiding the reuse of production code and focusing on comparing output values with hard-coded values.
+* KISS > DRY ("Keep it simple, Stupid!" and "Don't repeat yourself!")
+* Test close to production by focusing on testing a complete vertical slide and avoiding in-memory databases.
+* JUnit5 and AssertJ are a very good choice.
```

Comment:
Tbh I don't know if we need to mention those two here, since the project already requires tests to use AssertJ for example. I also don't know what is meant by "avoiding in-memory databases". Maybe we can drop that too since we have specific database utils?

Reply:
you are right I guess